### PR TITLE
chore: fix feature enabled

### PR DESCRIPTION
**Pull request by sighphyre**
Yggdrasil core has moved on from "feature_enabled" to featureEnabled. That's not publicly exposed in Edge and the SDKs can deal with it on their own terms while we migrate forward to a consistent casing scheme.

Because this affects the JSON serialising, this affects consuming SDKs as well, in practice that's just Ruby and .NET. Ruby doesn't need a patch because it already uses snake casing conversions. This brings .NET to the table

### CODE REVIEW

File: dotnet-engine/Yggdrasil.Engine.Tests/YggdrasilEngineTest.cs [hunk 1]
```diff
@@ -6,7 +6,25 @@
...
+        Payload = payload;
+        Enabled = enabled;
+        FeatureEnabled = featureEnabled;
+    }
+
+    public string Name { get; set; }
+    public Payload? Payload { get; set; }
+    public bool Enabled { get; set; }
+
+    [JsonPropertyName("feature_enabled")]
```
Comment (daveleek) [inline dotnet-engine/Yggdrasil.Engine.Tests/YggdrasilEngineTest.cs:25]: This one seems to be missing in the Variant class?

File: dotnet-engine/Yggdrasil.Engine/Types.cs [hunk 2]
```diff
@@ -29,22 +29,20 @@ public class EngineResponse<TValue> : EngineResponse
...
+        FeatureEnabled = featureEnabled;
     }
 
     public static readonly Variant DISABLED_VARIANT = new Variant("disabled", null, false, false);
 
     public string Name { get; set; }
     public Payload? Payload { get; set; }
-    [JsonPropertyName("enabled")]
     public bool Enabled { get; set; }
-    [JsonPropertyName("feature_enabled")]
```
Comment (daveleek) [inline dotnet-engine/Yggdrasil.Engine/Types.cs:46]: Shouldn't we keep this if we add it to the test case?

**Review by daveleek — COMMENTED** [review]
Question around a removed JsonPropertyName

File: dotnet-engine/Yggdrasil.Engine.Tests/YggdrasilEngineTest.cs (hunk 1 above)
Comment (sighphyre) [reply dotnet-engine/Yggdrasil.Engine.Tests/YggdrasilEngineTest.cs:25]: Yes, by design, the Variant class can and should handle "featureEnabled", this class only exists so we can materialise the "feature_enabled" property in the spec tests

File: dotnet-engine/Yggdrasil.Engine/Types.cs (hunk 2 above)
Comment (sighphyre) [reply dotnet-engine/Yggdrasil.Engine/Types.cs:46]: Nah, wanna yet it. The point of this PR to stop dealing with "feature_enabled" and deal with "featureEnabled" instead so we can get consistent casing

**Review by daveleek — APPROVED** [review]
Fair enough, ship it! :)

**Pull request merged**